lib/compress/zstd_compress.c

```diff
@@ -1390,9 +1389,31 @@ ZSTD_reset_matchState(ZSTD_matchState_t* ms,
     }
 
     /* table Space */
-    DEBUGLOG(4, "reset table : %u", crp!=ZSTDcrp_noMemset);
+    DEBUGLOG(4, "reset table : %u", crp!=ZSTDcrp_noMemset && (crp == ZSTDcrp_forceMemset || cParams->strategy != ZSTD_fast));
     assert(((size_t)ptr & 3) == 0);  /* ensure ptr is properly aligned */
+#ifdef NORESET
```

Comment:
You may want to separate testing for correctness, which this section covers,
and testing for speed / benchmark, since this section "gets in the way" by consuming CPU to fill the table with noise.

Reply:
For correctness I have simply ran 'make test'. As well, check that with and without NORESET the compressed output of the files I used for benchmarking are identical.

Once we understand what/if additional benchmarking needs to be done, I will remove the NORESET #define and as well remove the code that fills the table with noise.

I think the point of this task is to complete the change in zstd_fast.c, verify its correctness and then benchmark it.